### PR TITLE
Pay screen

### LINKED ISSUE #261: Thalia Pay history
Josst would like to see how much money he's spent.

### PR DESCRIPTION
**Pull request by JeeVee11**
Closes #261.

### Summary
This adds a screen where users can see their outstanding Thalia pay payments.

### CODE REVIEW

File: test/mocks.mocks.dart [hunk 1]
```diff
@@ -1473,20 +1473,20 @@ class MockPaymentUserCubit extends _i1.Mock implements _i2.PaymentUserCubit {
...
+  //         this,
+  //         Invocation.getter(#state),
+  //       ),
+  //     ) as _i2.DetailState<_i3.PaymentUser>);
+  // @override
+  // _i5.Stream<_i2.DetailState<_i3.PaymentUser>> get stream =>
+  //     (super.noSuchMethod(
+  //       Invocation.getter(#stream),
+  //       returnValue: _i5.Stream<_i2.DetailState<_i3.PaymentUser>>.empty(),
+  //     ) as _i5.Stream<_i2.DetailState<_i3.PaymentUser>>);
```
Comment (JAicewizard) [inline test/mocks.mocks.dart:1489]: Are these neccesary? They should either be removed or uncommented

File: lib/api/concrexit_api_repository.dart [hunk 2]
```diff
@@ -1180,4 +1180,39 @@ class ConcrexitApiRepository implements ApiRepository {
...
+      int? offset,
+      String? ordering,
+      DateTime? start,
+      DateTime? end,
+      List<PaymentType>? type,
+      bool? settled}) {
+    return sandbox(() async {
+      const paymentTypeMap = {
+        PaymentType.tpayPayment: 'tpay_payment',
+        //PaymentType.cardPayment: 'card_payment'
```
Comment (JAicewizard) [inline lib/api/concrexit_api_repository.dart:1196]: Remove this

**Review by JAicewizard — CHANGES_REQUESTED** [review]

**Comment by JAicewizard** [comment]
Very nice, I will try it out tomorrow.
We will have to figure out what causes the 500 first before merging, but that's for the concrexit team to figure out.

**Comment by JeeVee11** [comment]
> Very nice, I will try it out tomorrow. We will have to figure out what causes the 500 first before merging, but that's for the concrexit team to figure out.

So you think it is at the concrexit side? Or is the app making a bad request or something?

File: test/mocks.mocks.dart [hunk 3]
```diff
@@ -1501,23 +1501,23 @@ class MockPaymentUserCubit extends _i1.Mock implements _i2.PaymentUserCubit {
...
+  //     );
+  // @override
+  // void onChange(_i6.Change<_i2.DetailState<_i3.PaymentUser>>? change) =>
+  //     super.noSuchMethod(
+  //       Invocation.method(
+  //         #onChange,
+  //         [change],
+  //       ),
+  //       returnValueForMissingStub: null,
+  //     );
```
[the remark this answers is not in the document]
Comment (DeD1rk) [reply test/mocks.mocks.dart]: Probably just need to regenerate generated files

**Comment by DeD1rk** [comment]
The 500 is indeed a website thing, I'm not exactly sure why it _changed_ but a middleware is failing to set request.member in API requests, and a simple fix is to instead use only request.user. But I would like to know what broke it

**Comment by DeD1rk** [comment]
I'll probably fix it tomorrow

**Comment by DeD1rk** [comment]
I've got a fix in https://github.com/svthalia/concrexit/pull/2846

**Comment by JeeVee11** [comment]
Nice!

**Comment by JeeVee11** [comment]
OK it works now.

**Comment by JAicewizard** [comment]
Looks good, but tests should be fixed

File: android/app/src/main/kotlin/com/reaxit/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.reaxit
+
+import io.flutter.embedding.android.FlutterActivity
+
+class MainActivity: FlutterActivity() {
+}
```
Comment (JAicewizard) [inline android/app/src/main/kotlin/com/reaxit/MainActivity.kt:6]: Is this neccesary?

File: lib/ui/widgets/pay_tile.dart [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            widget.payment.topic,
+            maxLines: 2,
+            overflow: TextOverflow.ellipsis,
+          ),
+          trailing: Text(
+            '€ ${widget.payment.amount}',
+            maxLines: 1,
+            overflow: TextOverflow.ellipsis,
+          ),
+        )),
```
Comment (JAicewizard) [inline lib/ui/widgets/pay_tile.dart:58]: Add comma in between

**Review by JAicewizard — CHANGES_REQUESTED** [review]
Couple small changes but it looks good. On staging some payments have no data in the extra section, but I'm sure that thats just because they're not present if manually added.

**Review by JAicewizard — APPROVED** [review]

**Pull request merged**